lib/pengine/unpack.c

```diff
@@ -3568,15 +3593,23 @@ remap_operation(xmlNode *xml_op, pe_resource_t *rsc, const pe_node_t *node,
                          "on %s at %s as failure",
                          *rc, task, rsc->id, pe__node_name(node),
                          last_change_s);
-                *status = PCMK_EXEC_ERROR;
+                remap_because(status, &why, PCMK_EXEC_ERROR,
+                              "unknown exit status");
             }
             break;
     }
 
     free(last_change_s);
 
-    pe_rsc_trace(rsc, "Remapped %s status to '%s'",
-                 key, pcmk_exec_status_str(*status));
+remap_done:
+    if (why != NULL) {
+        pe_rsc_trace(rsc,
+                     "Remapped %s result from [%s: %s] to [%s: %s] "
+                     "because of %s",
+                     key, pcmk_exec_status_str(orig_exec_status),
+                     crm_exit_str(orig_exit_status),
+                     pcmk_exec_status_str(*status), crm_exit_str(*rc), why);
```

Comment:
Exit status is a little unusual in that we have an enum, but we always define it as int, because the ultimate sources of the value often don't conform to our model and return an int, plus int lets us convert back and forth between different representations. Worth a comment.